modules/control-utility/src/test/java/org/apache/ignite/util/MetricCommandTest.java

```diff
@@ -0,0 +1,644 @@
+    @Test
+    public void testNodeIdMissedFailure() {
+        assertContains(log, executeCommand(EXIT_CODE_INVALID_ARGUMENTS, CMD_METRIC, SYS_METRICS, NODE_ID.argName()),
+            "ID of the node from which metric values should be obtained is expected.");
+    }
+
+    /** Tests command error output in case value of {@link MetricCommandArg#NODE_ID} argument is invalid.*/
+    @Test
+    public void testInvalidNodeIdFailure() {
+        assertContains(log,
+            executeCommand(EXIT_CODE_INVALID_ARGUMENTS, CMD_METRIC, SYS_METRICS, NODE_ID.argName(), "invalid_node_id"),
+            "Failed to parse " + NODE_ID.argName() +
+                " command argument. String representation of \"java.util.UUID\" is exepected." +
+                " For example: 123e4567-e89b-42d3-a456-556642440000"
+        );
+    }
+
+    /** Tests command error output in case multiple metric names are specified. */
+    @Test
+    public void testMultipleMetricNamesFailure() {
+        assertContains(log,
+            executeCommand(EXIT_CODE_INVALID_ARGUMENTS, CMD_METRIC, IGNITE_METRICS, SYS_METRICS),
+            "Multiple metric(metric registry) names are not supported.");
+    }
+
+    /** Tests command error output in case {@link MetricCommandArg#NODE_ID} argument value refers to nonexistent node. */
+    @Test
+    public void testNonExistentNodeIdFailure() {
+        String incorrectNodeId = UUID.randomUUID().toString();
+
+        assertContains(log,
+            executeCommand(EXIT_CODE_INVALID_ARGUMENTS, CMD_METRIC, "--node-id", incorrectNodeId, IGNITE_METRICS),
+            "Failed to perform operation.\nNode with id=" + incorrectNodeId + " not found");
+    }
+
+    /** Tests command output in case nonexistent metric name is specified. */
+    @Test
+    public void testNonExistentMetric() {
+        assertContains(log, executeCommand(EXIT_CODE_OK, CMD_METRIC, IGNITE_METRICS + SEPARATOR),
+            "No metric with specified name was found [name=" + IGNITE_METRICS + SEPARATOR + ']');
+
+        assertContains(log, executeCommand(EXIT_CODE_OK, CMD_METRIC, "nonexistent.metric"),
+            "No metric with specified name was found [name=nonexistent.metric]");
+    }
+
+    /** */
+    @Test
+    public void testCommunicationMetrics() {
+        Map<String, String> metrics = metrics(ignite0, COMM_METRICS);
+
+        assertTrue(Integer.parseInt(metrics.get(metricName(COMM_METRICS, "SentBytesCount"))) > 0);
+        assertTrue(Integer.parseInt(metrics.get(metricName(COMM_METRICS, "ReceivedBytesCount"))) > 0);
+        assertEquals("0", metrics.get(metricName(COMM_METRICS, "OutboundMessagesQueueSize")));
+        assertTrue(Integer.parseInt(metrics.get(metricName(COMM_METRICS, "ReceivedMessagesCount"))) > 0);
+        assertTrue(Integer.parseInt(metrics.get(metricName(COMM_METRICS, "SentMessagesCount"))) > 0);
+    }
+
+    /** */
+    @Test
+    public void testJobsMetrics() {
```

Comment:
Le'ts keep one test for each type of metric and remove all other tests.

Reply:
Done.